### PR TITLE
Add stackdriver tracer support to mesh config

### PR DESCRIPTION
**Pull request by mandarjog**
1. Add debug param as it is relevant to testing.
2. Add 3 other config params which have 200 as defaults. We do need these to performance test and come up with potentially better default.
3. Left out all other params including project_id which we will get from the cluster.

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -66,6 +66,22 @@ message Tracing {
...
   }
 
+  // Stackdriver defines configuration for a Stackdriver tracer.
+  // See [Opencensus trace config](https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/trace/v1/trace_config.proto) for details.
+  message Stackdriver {
+    // debug enables trace output to stdout.
+    bool debug = 1;
+    // The global default max number of attributes per span.
+    // default is 200.
+    int64 max_number_of_attributes = 2;
```
Comment (kyessenov) [inline mesh/v1alpha1/proxy.proto]: per proto semantics, should be google.protobuf.IntValue instead.

Comment (douglas-reid) [inline mesh/v1alpha1/proxy.proto]: do we expect these to be set or ever configured by an operator for their mesh?

Comment (douglas-reid) [reply mesh/v1alpha1/proxy.proto]: Let's mark these fields as "hidden", if we can. With that, this would lgtm.

**Comment by objectiser** [comment]
@mandarjog agree this is a better approach - only thing is whether the additional config fields, such as max num of attributes, should be in a separate message defn - if they are applicable to any OC exporter?

**Comment by kyessenov** [comment]
The limits are applicable to all exporters I think, so it makes sense to pull them out of stackdriver.
On the other hand, we should consider how to enable zipkin tracing through opencensus via this API (is that a bool `use_opencensus_for_zipkin`, or another tracer?).

**Comment by objectiser** [comment]
It would be worth experimenting with both zipkin configs to see whether the data being reported as significantly different - hopefully not. In which case ideally we could just switch.

However I believe there will be one issue at the moment - with the current zipkin tracer is can be instructed not to sample the current trace instance (assuming a root span) when a health check filter is defined. The OpenCensus tracer is not currently capable of changing the sampling decision after the span has already been created.

Would be good to get a fix for this at some point, as it also means stack driver will be receiving trace instances for health checks aswell.

**Comment by kyessenov** [comment]
@objectiser do you mind filing an issue for that and CC me? We can see if this can be fixed in either envoy or opencensus. I fixed the span name to use the route decorator recently, so that should make it into envoy soon, once the conflicting googleapis shebang is resolved.

**Comment by objectiser** [comment]
@kyessenov sure will do it tomorrow.

**Comment by objectiser** [comment]
https://github.com/envoyproxy/envoy/issues/7422

**Comment by objectiser** [comment]
@kyessenov The other thing that would make migrating from the current envoy/zipkin tracer to opencensus zipkin exporter easier would be if OC/C++ supported the B3 propagation format (both separate and single header variants). Ideally the tracer could then be configured for inbound/outbound B3 and TraceContext propagation by default.

**Comment by mandarjog** [comment]
@kyessenov @douglas-reid updated per comments, please check.

**Pull request merged**